docs/layouts.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Blazor layouts
+author: rstropek
+description: Learn how to create reusable layout components for Blazor apps.
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 04/09/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: client-side/blazor/layouts
+---
+# Blazor layouts
+
+By [Rainer Stropek](https://www.timecockpit.com)
+
+[!INCLUDE[](~/includes/blazor-preview-notice.md)]
+
+Blazor apps typically contain more than one page. Layout elements, such as menus, copyright messages, and logos, must be present on all pages. Copying the code of these layout elements onto all of the pages of an app isn't an efficient solution. Such duplication is hard to maintain and probably leads to inconsistent content over time. *Blazor layouts* solve this problem.
+
+## What are Blazor layouts?
+
+Technically, a layout is just another Blazor component. A layout is defined in a Razor template or in C# code and can contain data binding, dependency injection, and other ordinary features of components. Two additional aspects are special and turn a Blazor component into a Blazor layout:
+
+* The layout component must implement `Microsoft.AspNetCore.Blazor.Layouts.ILayoutComponent`. This interface adds a property `Body` to the component, which contains the content to be rendered inside the layout.
+* The layout component contains the Razor keyword `@Body`. During rendering, it's replaced by the content of the layout.
+
+The following code sample shows the Razor template of a layout component. Note the use of `ILayoutComponent` and `@Body`:
+
+```csharp
+@implements ILayoutComponent
+
+<header>
+    <h1>ERP Master 3000</h1>
+    <p>Current user: @UserName</p>
```

Comment:
It might be better not to have the `@UserName` here (or the property below) because it raises the question of "How do I populate the `UserName` value?" and this document doesn't answer that. Nor do I think there is a good answer right now.

Reply:
Great feedback, thank you. I will change that in the course of the day.